File: webmvc/rest-server/src/main/java/com/example/web/UserRequest.java
package com.example.web;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

import org.springframework.beans.BeanUtils;

import com.example.domain.Gender;
import com.example.domain.User;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserRequest {
    private @NotEmpty String name;
    private @NotNull Gender gender;
    private String card;

    public User toUser() {
        User user = new User();
        BeanUtils.copyProperties(this, user);
        return user;
    }
}

File: others/scala/src/main/scala/com/example/web/UserRequest.scala
package com.example.web

import javax.validation.constraints.NotEmpty
import org.hibernate.validator.constraints.Range

import scala.annotation.meta.field
import scala.beans.BeanProperty

case class UserRequest(@BeanProperty
                       @(NotEmpty@field)
                       name: String,

                       @BeanProperty
                       @(Range@field)(min = 20, max = 60)
                       age: Byte) {

  def this() = this(null, 0)
}
